ability to switch between big & small db dump #121

kimaina opened the pull request:
<!-- This is based on openmrs-cord PR template. -->
## Description of what I changed
I have modified `openmrs-compose.yaml` to accommodate for #121 
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->
<!--- If your PR is related to an issue , please mention it here. -->
<!--- It is generally a good practice to first file an issue with enough
  context and reference it in the PR, but if you don't have that, please remove
  this section. -->

## E2E test
<!-- There are different scenarios for using the tools in this repo; please 
  help your reviewers by describing how you have e2e tested your change. -->
TESTED:
OMRS Environmental PR, as such I tested: docker-compose -f openmrs-compose.yaml up
## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I am familiar with Google Style Guides for the language I have coded in.

  No? Please take some time and review [Java](https://google.github.io/styleguide/javaguide.html) and [Python](https://google.github.io/styleguide/pyguide.html) style guides. Note, when in conflict, OpenMRS style guide overrules.

- [ ] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

mozzy11 commented on .env:
```diff
@@ -13,4 +13,5 @@
-COMPOSE_HTTP_TIMEOUT=200
\ No newline at end of file
+COMPOSE_HTTP_TIMEOUT=200
+DATABASE_DUMP_PATH=./utils/dbdump # for big data dump use ./utils/big-dbdump
```
@kimaina ,thanks here,
but it looks like its same thing as leaving `# for big data dump use   ./utils/big-dbdump ` directly in the compose file. 
i dont see any advantage of using an env variable

bashir2 commented on utils/big-dbdump/create_db.sql:
```diff
@@ -0,0 +1,43 @@
+-- Copyright 2020 Google LLC
```
Is this a symlink to the other copy?

bashir2 reviewed (commented):
Thanks @kimaina; just a minor comment/question.
 
@mozzy11 re. your comment [here](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/pull/122#discussion_r578383613) the one benefit that I see is that when I want to switch between the two DBs I don't need to change source files (just an env. var). Changing source files will keep bugging me when I make git commits.

bashir2 commented:
@kimaina, now that PR #118 is merged please make sure the new `.sql` file is also copied to `big-dbdump/`. Another approach in lieu of copying is to add a `.sh` script in `dbdump/` and feed the right `.sql` files into the MySQL DB at initialization (so instead of using all `.sql` files, we decide based on an env. var.). It seems that the entry-point of MySQL docker image [supports this](https://github.com/mysql/mysql-docker/blob/b4405db9783ca814c8ac6e068c0826b803f49c2d/5.7/docker-entrypoint.sh#L151). That way we can also hard-code the order in which we want these `.sql` files to be executed, instead of relying on the alphabetic order; and we should rename `.sql` files too (@mozzy11 FYI).

Anyways, up to you which was you prefer.

kimaina commented:
> @kimaina, now that PR #118 is merged please make sure the new `.sql` file is also copied to `big-dbdump/`. Another approach in lieu of copying is to add a `.sh` script in `dbdump/` and feed the right `.sql` files into the MySQL DB at initialization (so instead of using all `.sql` files, we decide based on an env. var.). It seems that the entry-point of MySQL docker image [supports this](https://github.com/mysql/mysql-docker/blob/b4405db9783ca814c8ac6e068c0826b803f49c2d/5.7/docker-entrypoint.sh#L151). That way we can also hard-code the order in which we want these `.sql` files to be executed, instead of relying on the alphabetic order; and we should rename `.sql` files too (@mozzy11 FYI).
> 
> Anyways, up to you which was you prefer.

sounds like a nice idea! Let's use this approach!

bashir2 commented on utils/dbdump/atomfeed_db_sql:
```diff
@@ -13,8 +13,8 @@
 -- This is to create a database for Atom Feed Client of OpenMRS.
--- To use this, run: mysql --user=[USER] --password=[PASSWORD] < create_db.sql
--- Or use this for a password prompt: mysql --user=[USER] -p < create_db.sql
+-- To use this, run: mysql --user=[USER] --password=[PASSWORD] < atomfeed_db.sql
```
nit: replace `.sql` with `_sql` ditto next line

kimaina replied:
fixed this, thanks!

bashir2 commented on utils/dbdump/init.sh:
```diff
@@ -0,0 +1,32 @@
+# This script is executed at run time during container initialization / entrypoint
+echo "[Entrypoint] Database dump started";
+echo "[Entrypoint] Databse dump mode: $DATABASE_DUMP_MODE";
+if [ "$DATABASE_DUMP_MODE" == "big" ]; then
```
nit: prefer `[[]]` over `[]` ([style guide](https://google.github.io/styleguide/shellguide.html#s6.3-tests))

kimaina replied:
fixed this, thanks!

bashir2 commented on utils/dbdump/init.sh:
```diff
@@ -0,0 +1,32 @@
+echo "[Entrypoint] Database dump started";
+echo "[Entrypoint] Databse dump mode: $DATABASE_DUMP_MODE";
+if [ "$DATABASE_DUMP_MODE" == "big" ]; then
+      echo "[Entrypoint] dumping openmrs_big_sql";
```
nit: Please use 2-space indent ([style guide](https://google.github.io/styleguide/shellguide.html#s5.1-indentation))

kimaina replied:
fixed this, thanks!

bashir2 reviewed (approved):
Thanks @kimaina for making the changes; all comments are minor, hence approving.
Do you know why did [this test](https://travis-ci.com/github/GoogleCloudPlatform/openmrs-fhir-analytics/jobs/485042662) fail? Please feel free to merge this after addressing the nits and making sure the e2e test passes.

kimaina commented:
> Thanks @kimaina for making the changes; all comments are minor, hence approving.
> Do you know why did [this test](https://travis-ci.com/github/GoogleCloudPlatform/openmrs-fhir-analytics/jobs/485042662) fail? Please feel free to merge this after addressing the nits and making sure the e2e test passes.

Thanks @bashir2, I think it is failing because OMRS did not / took a long time to start: Here is the last log-->
`
E2E: BATCH MODE TEST: TERMINATING TEST AS OPENMRS TOOK TOO LONG TO START`
